src/main/java/htsjdk/samtools/reference/AbstractFastaSequenceFile.java

```diff
@@ -82,32 +73,41 @@
     AbstractFastaSequenceFile(final Path path, final String source, final SAMSequenceDictionary sequenceDictionary) {
         this.path = path;
         this.source = source;
-        this.sequenceDictionary = sequenceDictionary;
+        this.dictionary = new Lazy<>(() -> sequenceDictionary);
     }
 
-    protected static File findSequenceDictionary(final File file) {
-        final Path dictionary = findSequenceDictionary(IOUtil.toPath(file));
-        if (dictionary == null) {
-            return null;
+    /** Attempts to find and load the sequence dictionary if present. */
+    protected SAMSequenceDictionary findAndLoadSequenceDictionary(final Path fasta) {
+        final Path dictPath = findSequenceDictionary(path);
+        if (dictPath == null) return null;
+
+        IOUtil.assertFileIsReadable(dictPath);
+        try (InputStream dictionaryIn = IOUtil.openFileForReading(dictPath)) {
+            return ReferenceSequenceFileFactory.loadDictionary(dictionaryIn);
+        }
+        catch (Exception e) {
+            throw new SAMException("Could not open sequence dictionary file: " + dictPath, e);
         }
-        return dictionary.toFile();
     }
 
-    protected static Path findSequenceDictionary(final Path path) {
-        if (path == null) {
+    /** Attempts to locate the sequence dictionary file adjacent to the reference fasta file. */
+    protected static Path findSequenceDictionary(final Path fastaPath) {
+        if (fastaPath == null) {
             return null;
         }
         // Try and locate the dictionary with the default method
-        final Path dictionary = ReferenceSequenceFileFactory.getDefaultDictionaryForReferenceSequence(path); path.toAbsolutePath();
```

Comment:
weird pointless code there you discovered....

Reply:
Yeah.  +1 for IntelliJ graying out unused code.